src/Protobuf/Decode.elm

```diff
@@ -726,17 +777,50 @@ tagDecoder =
             )
 
 
-varIntDecoder : Decode.Decoder ( Int, Int )
-varIntDecoder =
-    Decode.unsignedInt8
-        |> Decode.andThen
-            (\octet ->
-                if Bitwise.and 0x80 octet == 0x80 then
-                    Decode.map (\( usedBytes, value ) -> ( usedBytes + 1, Bitwise.and 0x7F octet + Bitwise.shiftLeftBy 7 value )) varIntDecoder
+type alias DecodeVarInt intType config =
```

Comment:
Why have you introduced this `DecodeVarInt`? And why do all the functions use this one rather than the internal `IntType` that you have defined?